livox_ros2_driver/livox_ros2_driver/lddc.cpp

```diff
@@ -712,433 +702,385 @@ void Lddc::initializeDiagnostics()
+  std::string error_str = "OK";
 
-    if (device_info == nullptr) {
-      stat.add(broadcast_code, "disconnected");
-      continue;
-    }
+  std::pair<std::string, DeviceInfo*> lidar;
+  getLidarByBroadcastcode(lidar, broadcast_code);
 
-    if (device_info->state == kLidarStateInit) {
-      stat.addf(broadcast_code, "%d%%", device_info->status.progress);
-      continue;
-    }
+  const auto & device_info = lidar.second;
 
+  if (device_info == nullptr) {
+    error_str = "disconnected";
+  } else if (device_info->state == kLidarStateInit) {
+    error_str = fmt::format("progress: %d%%", device_info->status.progress);
+  } else {
     VoltageStatus status = static_cast<VoltageStatus>(
       device_info->status.status_code.lidar_error_code.volt_status);
     if (status == VoltageStatus::High) {
       level = DiagStatus::WARN;
     } else if (status == VoltageStatus::ExtremelyHigh) {
       level = DiagStatus::ERROR;
     }
-
-    stat.add(broadcast_code, motor_dict_.at(level));
-    whole_level = std::max(whole_level, level);
+    error_str = voltage_dict_.at(level);
   }
 
-  stat.summary(whole_level, voltage_dict_.at(whole_level));
+    stat.add(broadcast_code, error_str);
+    stat.summary(level, error_str);
 }
 
-void Lddc::checkMotor(diagnostic_updater::DiagnosticStatusWrapper & stat)
+void Lddc::checkMotor(diagnostic_updater::DiagnosticStatusWrapper &stat,
+                      const std::string &broadcast_code)
 {
-  if (lidar_count_ == 0) {
-    stat.summary(DiagStatus::WARN, "No LiDARs Connected");
-    return;
-  }
-
-  int whole_level = DiagStatus::OK;
+  int level = DiagStatus::OK;
+  std::string error_str = "OK";
 
-  for (const auto & lidar : lds_->connected_lidars_) {
-    int level = DiagStatus::OK;
+  std::pair<std::string, DeviceInfo*> lidar;
+  getLidarByBroadcastcode(lidar, broadcast_code);
 
-    const auto & broadcast_code = lidar.first;
-    const auto & device_info = lidar.second;
-
-    if (device_info == nullptr) {
-      stat.add(broadcast_code, "disconnected");
-      continue;
-    }
-
-    if (device_info->state == kLidarStateInit) {
-      stat.addf(broadcast_code, "%d%%", device_info->status.progress);
-      continue;
-    }
+  const auto &device_info = lidar.second;
 
+  if (device_info == nullptr) {
+    error_str = "disconnected";
+  } else if (device_info->state == kLidarStateInit) {
+    error_str = fmt::format("progress: %d%%", device_info->status.progress);
+  } else {
     MotorStatus status = static_cast<MotorStatus>(
       device_info->status.status_code.lidar_error_code.motor_status);
     if (status == MotorStatus::Warning) {
       level = DiagStatus::WARN;
     } else if (status == MotorStatus::Error) {
       level = DiagStatus::ERROR;
     }
-
-    stat.add(broadcast_code, motor_dict_.at(level));
-    whole_level = std::max(whole_level, level);
+    error_str = motor_dict_.at(level);
   }
 
-  stat.summary(whole_level, motor_dict_.at(whole_level));
+  stat.add(broadcast_code, error_str);
```

Comment:
[Minor] `stat.add("broadcast_code", broadcast_code);` might be better because you split diagnostics.

Reply:
OK, I changed https://github.com/tier4/livox_ros2_driver/pull/14/commits/9dea16b0ac9c45c0457485833bac0d5be81f2529.